objects/src/notes/mod.rs

```diff
@@ -0,0 +1,96 @@
+use super::{assets::Asset, Digest, Felt, Hasher, NoteError, Vec, Word, WORD_SIZE, ZERO};
+
+mod script;
+pub use script::NoteScript;
+
+mod vault;
+pub use vault::NoteVault;
+
+// NOTE
+// ================================================================================================
+
+/// A note which can be used to transfer assets between accounts.
+///
+/// This struct is a full description of a note which is needed to execute a note in a transaction.
+/// A note consists of:
+/// - A set of assets stored in a vault.
+/// - A script which must be executed in a context of some account to claim the assets.
+/// - A serial number which can be used to break linkability between note hash and note nullifier.
+#[derive(Debug, Eq, PartialEq)]
+pub struct Note {
+    script: NoteScript,
+    vault: NoteVault,
+    serial_num: Word,
+}
+
+impl Note {
+    // CONSTRUCTOR
+    // --------------------------------------------------------------------------------------------
+    /// Returns a new note created with the specified parameters.
+    ///
+    /// # Errors
+    /// Returns an error if:
+    /// - Compilation of note script fails.
+    /// - The number of provided assets exceeds 1000.
+    /// - The list of assets contains duplicates.
+    pub fn new(script_src: &str, assets: &[Asset], serial_num: Word) -> Result<Self, NoteError> {
```

Comment:
Maybe use `AsRef<str>` for the script source? This way, we can put here strings, strs, chars, etc

Reply:
Agreed! Changed!